src/main/java/org/broadinstitute/hellbender/tools/walkers/CombineGVCFs.java

```diff
@@ -412,14 +412,17 @@ public Object onTraversalSuccess() {
             return null;
         }
 
-        SimpleInterval interval = prevPos != null ? new SimpleInterval(prevPos.getContig(), prevPos.getStart(), variantContextsOverlappingCurrentMerge.stream().map(VariantContext::getEnd).max(Comparator.naturalOrder()).get()) :
-                storedReferenceContext.getInterval();
-
-        createIntermediateVariants(interval);
-
-        // there shouldn't be any state left unless the user cut in the middle of a gVCF block
         if ( !variantContextsOverlappingCurrentMerge.isEmpty() ) {
-            logger.warn("You have asked for an interval that cuts in the middle of one or more gVCF blocks. Please note that this will cause you to lose records that don't end within your interval.");
+            // finish off the last blocks
+            final SimpleInterval lastInterval = new SimpleInterval(
```

Comment:
Hmmm... So this change is mostly agnostic and is definately important for the case where the assumption that the prevPos must be on the right contig (which wasn't checked before). I'm not convinced this works universally in the case where there is a gap before the last contig. So imagine there are still reads queued to be closed at prevPos, a gap in coverage, then a stack of variant contexts at the last position. It doesn't strike me that this should handle closing that last interval if it needs to generate its own stops (for example if BREAK_BANDS_LONG_NAME) is set then it may well generate breaks in the gap between prevpos and the last stack of VCs. I think we should prefer to use prevPos in these cases and only switch to variantContextsOverlappingCurrentMerge.get(0) in the case where prevpos is on the wrong contig/starts later than it should.

Reply:
Well, having thought about this a bit more, I'm not sure that problem is new to this change. As far as I can tell, this implementation of the tool only emits coverage for territory that is covered by the inputs, and doesn't pad out territory beyond that. In onTraversalSuccess, if prevPos is on a different contig than the remaining variants to be merged, we won't close out (pad out) the previous contig, but I don't think the tool ever does that. We might want to fix that, but I think thats a separate/bigger issue ? Otherwise, the variants that are remaining are all guaranteed to be on the same contig (right ?)  Sorry if I'm missing something, but please push back if you think I am.

Reply:
You know, you are right. I was thinking of the code being like a normal `apply()` call where you are holding onto something "next." Your fix looks good.